### PR TITLE
Add Entry: HTML Entities

### PR DESCRIPTION
**Pull request by SSwiniarski**
### Description

Add entry for HTML Entities, prompted by the Target Keyword Sheet.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

File: content/html/concepts/entities/entities.md
Comment (KTom101) [inline content/html/concepts/entities/entities.md:15]: including a little more on why entities would be used
```suggestion
Entities are character sequences placed in the text of an HTML document to print out reserved or other special characters which may otherwise be interpreted as HTML code by a browser. When parsing the HTML, a browser replaces the entity's character sequence with the character it represents.
```

**Review by KTom101 — COMMENTED** [review]
@SSwiniarski I've made very minor suggestions. This is a very solid entry!

**Comment by SSwiniarski** [comment]
@KTom101, I incorporated your suggestions along with some additional edits. Mostly to clarify the relation between entities and the unicode character set.

**Review by KTom101 — APPROVED** [review]
@SSwiniarski I like the changes. @Dusch4593 and @yangc95, this is ready for a second review when you are.

File: content/html/concepts/entities/entities.md
Comment (Dusch4593) [inline content/html/concepts/entities/entities.md:19]:
```suggestion
Entities are case-sensitive and can take one of three forms.
```

File: content/html/concepts/entities/entities.md
Comment (Dusch4593) [inline content/html/concepts/entities/entities.md:56]:
```suggestion
Here is an [extensive HTML 5 entity reference](https://dev.w3.org/html5/html-author/charref).
```

File: content/html/concepts/entities/entities.md
Comment (Dusch4593) [inline content/html/concepts/entities/entities.md:60]: I think this section could use some code snippets.

```
There is an entity `&Tab;` (or `&#9;`) which nominally produces a tab character. However, it will not behave as expected because HTML collapses whitespace where it appears in text with the exception of the `&nbsp;` entity. To render a tab in HTML text, it must be used within a `<pre>` tag to preserve the whitespace: 

[Code snippet with <pre> tag here.]

Alternatively, four `&nbsp;` entities can be used:

[Code snippet with 4 `&nbsp;`s here.]
```

File: content/html/concepts/entities/entities.md
Comment (Dusch4593) [inline content/html/concepts/entities/entities.md:54]:
```suggestion
| Entity Name | 1st Form | 2nd Form | 3rd Form |
| ------ | ------ | ------ | ------ |
| Non-breaking space | `&nbsp;` | `&#160;` | `&#x000A0;` |
| Less-than sign `<` | `&lt;` | `&#60;` | `&#x0003C;` |
| Greater-than sign `>` | `&gt;` | `&#62;` | `&#x0003E;` |
| Ampersand `&` | `&amp;` | `&#38;` | `&#x00026;` |
| Quote `"` | `&quot;` | `&#34;` | `&#x00022;` |
| Apostrophe | `&apos;` | `&#39;` | `&#x00027;` |
| Copyright `©` | `&copy;` | `&#169;` | `&#169;` |
| Degree `°` | `&deg;` | `&#176;` | `&#x000B0;` |
| Euro `€` | `&euro;` | `&#8364;` | `&#x020AC;` |
| Bullet character `•` | `&bullet;` | `&#8226;` | `&#x02022;` |
```

I'm wondering if this sort of information is better sorted under [a markdown table](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-tables)? This way we could include the 3rd form for each entity.

File: content/html/concepts/entities/entities.md
Comment (Dusch4593) [inline content/html/concepts/entities/entities.md:36]:
```suggestion
&#xnumber;
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@SSwiniarski Finished with my review and left some comments/suggestions.

**Comment by SSwiniarski** [comment]
@Dusch4593, I applied all your suggestions and made a few tweaks.

**Comment by Dusch4593** [comment]
@SSwiniarski @KTom101 anything left to check or consider before we merge this one?

File: content/html/concepts/entities/entities.md
Comment (yangc95) [inline content/html/concepts/entities/entities.md:15]:
```suggestion
Entities are character sequences placed in the text of an HTML document to print out reserved characters which may otherwise be interpreted as HTML code by a browser, or other special characters that would be difficult to generate with a typical keyboard. In fact, using reference numbers, an entity can represent any Unicode character. When parsing the HTML, a browser replaces the entity's character sequence with the character it represents.
```

File: content/html/concepts/entities/entities.md
Comment (yangc95) [inline content/html/concepts/entities/entities.md:62]:
```suggestion
There is an entity `&Tab;` (or `&#9;`) which nominally produces a tab character. However, it will not behave as expected because HTML collapses whitespace where it appears in text with the exception of the fixed space entity, `&nbsp;`. So to render a tab in HTML text, it needs to be used within a `<pre>` tag to preserve the whitespace.
```

**Comment by yangc95** [comment]
@SSwiniarski made some minor suggestions!

**Comment by SSwiniarski** [comment]
@yangc95, updated with your suggestions.

File: content/html/concepts/entities/entities.md
Comment (yangc95) [inline content/html/concepts/entities/entities.md:62]:
```suggestion
There is an entity `&Tab;` (or `&#9;`) which nominally produces a tab character. However, it will not behave as expected because HTML collapses whitespace where it appears in text with the exception of the non-breaking space entity, `&nbsp;`. So to render a tab in HTML text, it needs to be wrapped within a `<pre>` tag to preserve the whitespace.
```

**Review by yangc95 — APPROVED** [review]
Approving this, also added one last thing relating to the keywords!

Anything else before merging this? @KTom101 @Dusch4593

**Comment by Dusch4593** [comment]
@yangc95 everything LGTM! 👍

**Comment by KTom101** [comment]
@yangc95 Good to merge 👍

**Comment by SSwiniarski** [comment]
@yangc95, committed that last little update. Should be good to merge.

**Pull request merged**
